rpc/client.go

```diff
@@ -195,8 +194,10 @@ func (c *Client) Close() {
 	clientMu.Lock()
 	if !c.closed {
 		delete(clients, c.Addr().String())
+		c.mu.Lock()
 		c.healthy = false
 		c.closed = true
+		c.mu.Unlock()
```

Comment:
Looks like the comment next to the `Client.closed` field is inaccurate. That field is protected by `clientMu`. Mind moving the field to before `Client.mu` and indicate that it is protected by `clientMu`.